### ISSUE TITLE
UnusedVariable with run in with parameters

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/sonar-openedge#520]
Following code results in a false positive for the UnusedVariable rule.
The false positive only occurs when there are parameters specified in the run statement.

```
class UnusedInFalsePositive:

    method public void UsedHandle():
        define variable lProcedure as handle no-undo. // Variable lProcedure is assigned but never read  UnusedInFalsePositive.cls

        lProcedure = session:first-procedure.
        run test in lProcedure ("some parameter").
    end method.

    method public void UsedHandle2():
        define variable lProcedure as handle no-undo. // No problem here

        lProcedure = session:first-procedure.
        run testwithoutparameters in lProcedure.
    end method.

end class.
```

**Comment by gquerret** [comment]
Several changes were required for #507, and I think the `RUN ... IN ...` statement has already been taken care of. I'll double check. Fix will be included in next release

**Comment by dominickdm** [comment]
+1 for this issue

also "RUN ... ON  " is causing false positives
for example:`

`DEFINE PRIVATE VARIABLE AppServerSynchronous AS HANDLE NO-UNDO.`
...
`AppServerSynchronous = Connection:Synchronous.`
...
`RUN be/mips/glims/server/logisticsmanager/boxunpacking/UnpackBoxGateway
            SINGLE-RUN SET UnpackBoxDataAccessGatewaySync ON AppServerSynchronous.`

**Comment by gquerret** [comment]
@cverbiest Parser issue for the `RUN ... IN` problem, it currently recognizes `lProcedure("someParameter")` as the expression for the `IN` keyword (as if lProcedure was a function or method call). Will be fixed, similar problem was already fixed in dynamic-new statements for example.

@dominickdm SINGLE-RUN, SINGLETON added as keywords, and ON keyword now adds a reference to the variable. Will be included in next release

**Comment by gquerret** [comment]
Merged in develop
